src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/stash/StashApiClient.java

```diff
@@ -562,4 +570,16 @@ public Socket createSocket(Socket socket, String host, int port, boolean autoClo
       return this.getSSLContext().getSocketFactory().createSocket(socket, host, port, autoClose);
     }
   }
+
+  public static class StashApiException extends Exception {
```

Comment:
minor: we could make it into a RuntimeException to avoid having to update all the test signatures

Reply:
No, I believe a checked exception is more suitable here. Exceptions related to Stash communications should be caught inside the plugin, and not necessarily where they are thrown (so that the plugin can skip building the whole PR if it cannot read just one page of the PR comments). Checked exceptions guarantee that the exception is caught somewhere.

Besides, RuntimeException is for issues that can be prevented programmatically, e.g. for out of bounds access.

The issue is discussed here: https://stackoverflow.com/questions/2190161/

Reply:
Agree. 

> Exceptions related to Stash communications should be caught inside the plugin, and not necessarily where they are thrown (so that the plugin can skip building the whole PR if it cannot read just one page of the PR comments).

Could you add this to the exceptions javadoc?

Reply:
I've added a similar text